src/doc/trpl/guessing-game.md

```diff
@@ -0,0 +1,991 @@
+
+```rust,ignore
+    let secret_number = rand::thread_rng().gen_range(1, 101);
+
+    println!("The secret number is: {}", secret_number);
+```
+
+We use the `rand::thread_rng()` function to get a copy of the random number
+generator, which is local to the particular [thread][concurrency] of execution
+we’re in. Because we `use rand::Rng`’d above, it has a `gen_range()` method
+available. This method takes two arguments, and generates a number between
+them. It’s inclusive on the lower bound, but exclusive on the upper bound,
+so we need `1` and `101` to get a number between one and a hundred.
+
+[concurrency]: concurrency.html
+
+The second line just prints out the secret number. This is useful while
+we’re developing our program, so we can easily test it out. But we’ll be
+deleting it for the final version. It’s not much of a game if it prints out
+the answer when you start it up!
+
+Try running our new program a few times:
+
+```bash
+$ cargo run
+   Compiling guessing_game v0.1.0 (file:///home/you/projects/guessing_game)
+     Running `target/debug/guessing_game`
+Guess the number!
+The secret number is: 7
+Please input your guess.
+4
+You guessed: 4
+$ cargo run
+     Running `target/debug/guessing_game`
+Guess the number!
+The secret number is: 83
+Please input your guess.
+5
+You guessed: 5
+```
+
+Great! Next up: let’s compare our guess to the secret guess.
+
+# Comparing guesses
+
+Now that we’ve got user input, let’s compare our guess to the random guess.
+Here’s our next step, though it doesn’t quite work yet:
+
+```rust,ignore
+extern crate rand;
+
+use std::io;
+use std::cmp::Ordering;
+use rand::Rng;
+
+fn main() {
+    println!("Guess the number!");
+
+    let secret_number = rand::thread_rng().gen_range(1, 101);
+
+    println!("The secret number is: {}", secret_number);
+
+    println!("Please input your guess.");
+
+    let mut guess = String::new();
+
+    io::stdin().read_line(&mut guess)
+        .ok()
+        .expect("failed to read line");
+
+    println!("You guessed: {}", guess);
+
+    match guess.cmp(&secret_number) {
+        Ordering::Less    => println!("Too small!"),
+        Ordering::Greater => println!("Too big!"),
+        Ordering::Equal   => println!("You win!"),
+    }
+}
+```
+
+A few new bits here. The first is another `use`. We bring a type called
+`std::cmp::Ordering` into scope. Then, four new lines at the bottom that use
+it:
+
+```rust,ignore
+match guess.cmp(&secret_number) {
+    Ordering::Less    => println!("Too small!"),
+    Ordering::Greater => println!("Too big!"),
+    Ordering::Equal   => println!("You win!"),
+}
+```
+
+The `cmp()` method can be called on anything that can be compared, and it
+takes a reference to the thing you want to compare it to. It returns the
+`Ordering` type we `use`d earlier. We use a [`match`][match] statement to
+determine exactly what kind of `Ordering` it is. `Ordering` is an
+[`enum`][enum], short for ‘enumeration’, which looks like this:
```

Comment:
Hm, this sentence is a bit confusing, since Ordering doesn't look like the enum below.